cli/src/program_v4.rs

```diff
@@ -1049,43 +1052,27 @@ fn send_messages(
                     ],
                     blockhash,
                 )?;
-                let result = rpc_client.send_and_confirm_transaction_with_spinner_and_config(
-                    &initial_transaction,
-                    config.commitment,
-                    config.send_transaction_config,
-                );
-                log_instruction_custom_error_ex::<SystemError, _>(
-                    result,
-                    &config.output_format,
-                    common_error_adapter,
-                )
-                .map_err(|err| format!("Account allocation failed: {err}"))?;
             } else {
                 return Err("Buffer account not created yet, must provide a key pair".into());
             }
-        } else if message.header.num_required_signatures == 2 {
-            // All other messages should require 2 signatures (payer, and authority)
-            let blockhash = rpc_client.get_latest_blockhash()?;
-
-            let mut initial_transaction = Transaction::new_unsigned(message.clone());
+        } else {
+            // All other messages have up to 2 signatures (payer, and authority).
             initial_transaction.try_sign(
                 &[config.signers[0], config.signers[*auth_signer_index]],
                 blockhash,
             )?;
-            let result = rpc_client.send_and_confirm_transaction_with_spinner_and_config(
-                &initial_transaction,
-                config.commitment,
-                config.send_transaction_config,
-            );
-            log_instruction_custom_error_ex::<SystemError, _>(
-                result,
-                &config.output_format,
-                common_error_adapter,
-            )
-            .map_err(|err| format!("Failed to send initial message: {err}"))?;
-        } else {
```

Comment:
Should we retain this check? i.e. would `num_required_signatures` be either 2 or 3, and all other values will return an error.